torchgeo/datasets/sentinel.py

```diff
@@ -25,6 +25,221 @@ class Sentinel(RasterDataset):
+        _(?P<band>[VH]{2})
+        \.
+    """
+    date_format = "%Y%m%dT%H%M%S"
+    all_bands = ["HH", "HV", "VV", "VH"]
+    separate_files = True
+
+    def __init__(
+        self,
+        root: str = "data",
+        crs: Optional[CRS] = None,
+        res: float = 10,
+        bands: Sequence[str] = ["VV", "VH"],
+        transforms: Optional[Callable[[Dict[str, Any]], Dict[str, Any]]] = None,
+        cache: bool = True,
+    ) -> None:
+        """Initialize a new Dataset instance.
+
+        Args:
+            root: root directory where dataset can be found
+            crs: :term:`coordinate reference system (CRS)` to warp to
+                (defaults to the CRS of the first file found)
+            res: resolution of the dataset in units of CRS
+                (defaults to the resolution of the first file found)
+            bands: bands to return (defaults to ["VV", "VH"])
+            transforms: a function/transform that takes an input sample
+                and returns a transformed version
+            cache: if True, cache file handle to speed up repeated sampling
+
+        Raises:
+            AssertionError: if ``bands`` is invalid
+            FileNotFoundError: if no files are found in ``root``
+        """
+        assert len(bands) > 0, "'bands' cannot be an empty list"
+        assert len(bands) == len(set(bands)), "'bands' contains duplicate bands"
+
+        for band in bands:
+            assert band in self.all_bands, f"invalid band '{band}'"
+
+        # Sentinel-1 uses dual polarization, it can only transmit either
+        # horizontal or vertical at a single time
+        msg = """
+'bands' cannot contain both horizontal and vertical transmit at the same time.
+To create a dataset containing both, use:
+
+    horizontal = Sentinel1(root, bands=["HH", "HV"])
+    vertical = Sentinel1(root, bands=["VV", "VH"])
+    dataset = horizontal | vertical
+"""
+        assert len({band[0] for band in bands}) == 1, msg
+
+        self.filename_glob = self.filename_glob.format(bands[0])
+
+        super().__init__(root, crs, res, bands, transforms, cache)
+
+    def plot(
+        self,
+        sample: Dict[str, Any],
+        show_titles: bool = True,
+        suptitle: Optional[str] = None,
+    ) -> plt.Figure:
+        """Plot a sample from the dataset.
+
+        Args:
+            sample: a sample returned by :meth:`RasterDataset.__getitem__`
+            show_titles: flag indicating whether to show titles above each panel
+            suptitle: optional string to use as a suptitle
+
+        Returns:
+            a matplotlib Figure with the rendered sample
+        """
+        bands = self.bands
+
+        if len(bands) == 1:
+            # Only horizontal or vertical receive, plot as grayscale
+
+            image = sample["image"][0]
+            image = torch.clamp(image, min=0, max=1)
+
+            title = f"({bands[0]})"
+        else:
+            # Both horizontal and vertical receive, plot as RGB
```

Comment:
Is there any reason to plot these images as RGB? The case to plot RGB images is rare as the third band ratio choice is extremely dependent on application. Why not just present an Nx1 subplot with all bands plotted in grayscale?

Reply:
Because RGB looks cooler 😄 

Seriously though, I only went with RGB cuz that's how we plot MSI datasets. When I view thumbnails of these images on Copernicus, they also choose to display an RGB image. What other third band ratio choices are common? We could add an option to select one or the other when plotting. Of course, users can always implement their own plotting code, this is just supposed to be a reasonable default.

Reply:
I'm not tied to displaying an image one way or another, but if it makes the code easier, then displaying N grayscale bands would be infinitely easier than dealing with all these combinations/ratios for RGB imagery.